pydantic_ai_slim/pydantic_ai/exceptions.py

```diff
@@ -158,6 +161,62 @@ def __init__(self, status_code: int, model_name: str, body: object | None = None
         super().__init__(message)
 
 
+class MCPError(RuntimeError):
+    """Base class for errors occurring during interaction with an MCP server."""
+
+    message: str
+    """The error message."""
+
+    def __init__(self, message: str):
+        self.message = message
+        super().__init__(message)
+
+    def __str__(self) -> str:
+        return self.message
+
+
+class MCPServerCapabilitiesError(MCPError):
+    """Raised when attempting to access server capabilities that aren't present."""
```

Comment:
Technically, we should do this for tools as well now. Let me know if you want me to raise a separate issue for this.